lms/static/js/utils/dropdown.js

```diff
@@ -0,0 +1,235 @@
+
+                            case 40:
+                                that.nextMenuItemLink(focused, menu);
+                                return false;
+                                break;
+
+                            case 27:
+                                that.closeDropdownMenus();
+                                break;
+                        }
+
+                    } else if(focused.is('.has-dropdown')) {
+
+                        switch(keyCode) {
+                            case 40:
+                                that.openDropdownMenu(focused);
+                                return false;
+                                break;
+
+                            case 13:
+                                that.openDropdownMenu(focused);
+                                break;
+
+                            case 27:
+                                that.closeDropdownMenus();
+                                break;
+                        }
+                    } else if (focused.is('.dropdown-menu')) {
+
+                        menu = focused.parent().parent();
+
+                        switch(keyCode) {
+                            case 40:
+                                that.focusFirstItem(menu);
+                                return false;
+                                break;
+
+                            case 27:
+                                that.closeDropdownMenus();
+                                break;
+                        }
+                    } else {
+
+                        switch(keyCode) {
+                            case 13:
+                            case 38:
+                            case 40:
+                            case 27:
+                        }
+                    }
+                });
+            },
+
+            previousMenuItemLink: function(focused, menu) {
+                var items = menu.children('.dropdown-item').find('.action'),
+                    items_count = items.length - 1,
+                    index = items.index(focused),
+                    prev = index - 1;
+
+                if (index === 0) {
+                    items.last().focus();
+                } else {
+                    items.eq(prev).focus();
+                }
+            },
+
+            nextMenuItemLink: function(focused, menu) {
+                var items = menu.children('.dropdown-item').find('.action'),
+                    items_count = items.length -1,
+                    index = items.index(focused),
+                    next = index + 1;
+
+                if (index === items_count) {
+                    items.first().focus();
+                } else {
+                    items.eq(next).focus();
+                }
+            },
+
+            focusFirstItem: function(menu) {
+                menu.find('.dropdown-item:first .action').focus();
+            },
+
+            closeDropdownMenu: function() {
+                var that = _dropdown;
```

Comment:
`that` here refers to the main object, and is used to reference the objects "variables" at the top.